src/helpers/pivot/pivot_presentation.ts

```diff
@@ -0,0 +1,570 @@
+          }
+        }
+      }
+
+      return mainDimension === "row" ? ranking : transpose2dPOJO(ranking);
+    }
+
+    private computeRunningTotal(
+      measure: PivotMeasure,
+      fieldNameWithGranularity: string,
+      mode: "running_total" | "%_running_total"
+    ): DomainGroups<number | undefined> {
+      const cellsRunningTotals: DomainGroups<number | undefined> = {};
+      const mainDimension = getFieldDimensionType(this, fieldNameWithGranularity);
+      const secondaryDimension = mainDimension === "row" ? "column" : "row";
+
+      let pivotCells = this.getPivotValueCells();
+
+      if (mainDimension === "column") {
+        // Transpose the pivot cells so we can do the same operations on the columns as on the rows
+        // This means that we need to transpose back the totals at the end
+        pivotCells = transposeMatrix(pivotCells);
+      }
+
+      for (const col of pivotCells) {
+        const colDomain = getDimensionDomain(this, secondaryDimension, col[0].domain);
+        const colDomainKey = domainToString(colDomain);
+        cellsRunningTotals[colDomainKey] = {};
+        const runningTotals: { [runningTotalKey: string]: number } = {};
+
+        const cellsWithValue = col
+          .map((cell) => ({
+            ...cell,
+            rowDomain: getDimensionDomain(this, mainDimension, cell.domain),
+            value: this.measureValueToNumber(
+              super.getPivotCellValueAndFormat(measure.name, cell.domain)
+            ),
+          }))
+          .filter((cell) => isFieldInDomain(fieldNameWithGranularity, cell.rowDomain));
+
+        for (const cell of cellsWithValue) {
+          const rowDomainKey = domainToString(cell.rowDomain);
+          const runningTotalKey = getRunningTotalDomainKey(
+            cell.rowDomain,
+            fieldNameWithGranularity
+          );
+
+          const runningTotal = (runningTotals[runningTotalKey] || 0) + cell.value;
+          runningTotals[runningTotalKey] = runningTotal;
+          cellsRunningTotals[colDomainKey][rowDomainKey] = runningTotal;
+        }
+
+        if (mode === "%_running_total") {
+          for (const cell of cellsWithValue) {
+            const rowDomain = cell.rowDomain;
+            const rowDomainKey = domainToString(rowDomain);
+            const runningTotalKey = getRunningTotalDomainKey(rowDomain, fieldNameWithGranularity);
+
+            const cellRunningTotal = cellsRunningTotals[colDomainKey][rowDomainKey] || 0;
+            const finalRunningTotal = runningTotals[runningTotalKey];
+            cellsRunningTotals[colDomainKey][rowDomainKey] = !finalRunningTotal
+              ? undefined
+              : cellRunningTotal / finalRunningTotal;
+          }
+        }
+      }
+      return mainDimension === "row" ? cellsRunningTotals : transpose2dPOJO(cellsRunningTotals);
+    }
+
+    private getGrandTotal(measure: string): number {
+      const grandTotal = super.getPivotCellValueAndFormat(measure, []);
```

Comment:
It doesn't seem to be a huge performance cost, but if we display the cells as % of grand total, we re-compute the grand total for each cell. We could probably add caching to the `SpreadsheetPivot`